_posts/2019/2019-07-24-how-to-transpile-node-modules-with-babel-and-webpack-in-a-monorepo.md

```diff
@@ -0,0 +1,140 @@
+
+### A story of Javascript standards and transpilation
+
+In this case though, we are not transpiling between Javascript and a derivative language, but between 2 standards of Javascript- ES5 and ES6.
+The ES stands for ECMAScript, which is a [scripting-language specification](https://en.wikipedia.org/wiki/ECMAScript){:target="_blank"}, created to standardize Javascript.
+ES5 refers to the fifth edition of ECMAScript, which is currently supported by all modern browsers.
+ES6- also known as ES2015, is the sixth edition and added several helpful features to Javascript which can be found [here](http://es6-features.org){:target="_blank"}.
+Now, to avoid browser compatibility issues, it is generally advisable to transpile any software meant to run in the browser and written in ES6 or greater, to ES5.
+This is because while most browsers have added support for ES6, there are still some holdouts(I'm looking at you Internet Explorer!!) with quite widespread usage.
+If you don't want to do this though, you can avoid, feature detect or polyfill browser or DOM apis not supported in IE11, or any browser for that matter.
+
+### Tools we use
+
+There are a number of tools for Javascript transpilation out there, but we use [Babel](https://babeljs.io/docs/en/){:target="_blank"} - an extremely powerful Javascript transpilation toolchain that supports conversion between Javascript standards as well as its derivative languages.
+Babel can be configured using a `.babelrc` file, or a `babel.config.js` file.
+Another tool we rely on heavily in our build environment is [Webpack](https://webpack.js.org/concepts){:target="_blank"}, a static module bundler for modern JavaScript applications.
+Webpack is highly configurable, and can be set up so one can configure Babel from within the webpack configuration file, which is usually called `webpack.config.js` using the Babel plugin.
+
+## So what happened then?
+
+We do not transpile the contents of our `node_modules` folder, to reduce the size of our bundle and avoid extremely long build times.
+This is because we expect that we receive already transpiled code from our dependencies.
+This time, we were wrong.
+
+## What we did...
+
+Of course, we cannot transpile our entire `node_modules` folder, since that would increase build time by a terrible factor.
+The compromise we came up with was to modify the regex pattern we used to exclude the `node_modules` folder in our webpack/babel config so that it excluded everything in the `node_modules` folder _except the folder containing the module that was not transpiled_.
+
+Essentially we changed this:
+
+```
+...
+  exclude: /node_modules/,
+...
+```
+
+to this:
+
+```
+...
+  exclude: /node_modules\/?!name-of-untranspiled-module.*/,
+...
+```
+
+## Why this didn't work...
+
+We should have expected this from the onset, but upon rebuilding the project we errored again on IE11, because _another module_ was also not transpiled!
```

Comment:
```suggestion
We should have expected this from the onset, but upon rebuilding the project we errored again on IE11, because _another module_ was also not being transpiled!
```